packages/fuels-abigen-macro/tests/test_projects/transaction_block_height/src/main.sw

```diff
@@ -0,0 +1,10 @@
+contract;
+
+abi MyContract {
+    fn test_function();
+}
+
+impl MyContract for Contract {
+    fn test_function() {
+    }
```

Comment:
There is a way but I've decided against returning the block height from the contract itself.

The reason is twofold:
1. Calling a contract function that will return the current block height will produce a block in and out of itself. This makes testing weird because you'd have these +2 increments due to the previous check and the actual call. And if you use the height getter contract function as the primary function to test maturity, then you'll have these weird cases where you need the thing to fail but also need to validate the height but can't since the contract failed.

2. Don't feel the need to be paranoid about the gql node API, it seems simple enough and we'd effectively be testing what the fuel-core team should test.

Reply:
> 1. Calling a contract function that will return the current block height will produce a block in and out of itself. 

@adlerjohn is this really the case? does a contract call that just reads some data get actually stored in the block as a transaction?

Reply:
Ok did it. There is a test now which confirms that block height from within sway and from the gql API is of the same value.

Not sure if we're the ones that should be testing this, but until a more generalized whole solution test suite gets implemented, I guess the SDK is the QA department :D